### ISSUE TITLE
Bug with Japanese language `ValueError: [E194] Unable to aligned mismatched text`

### ISSUE THREAD

**Issue by kovalevfm** [explosion/spaCy#6466]
## How to reproduce the behaviour
```python
import spacy
sp = spacy.load('ja_core_news_sm')
sp( 'あつつ( ̄^ ̄)ゞ')
>>> ValueError: [E194] Unable to aligned mismatched text 'あつつ( ̄^ ̄)ゞ' and words '['あ', 'つつ', '(', ' ', '̄', '^', ' ̄', ')', 'ゞ']'.
```

### Info about spaCy

* **spaCy version:** 2.3.2
* **Platform:** Linux-5.4.0-1029-gcp-x86_64-with-Ubuntu-18.04-bionic
* **Python version:** 3.6.9

```
>> python3 -m pip show ja_core_news_sm
Name: ja-core-news-sm
Version: 2.3.2
Summary: Japanese multi-task CNN trained on UD_Japanese-GSD v2.6-NE. Assigns context-specific token vectors, POS tags, dependency parses and named entities.
Home-page: https://explosion.ai
Author: Explosion and Megagon Labs Tokyo
Author-email: contact@explosion.ai
License: CC BY-SA 4.0
Location: /usr/local/virtualenv/tf1/lib/python3.6/site-packages
Requires: sudachidict-core, sudachipy, spacy
Required-by:
```
Stack trace:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/usr/local/virtualenv/tf1/lib/python3.6/site-packages/spacy/lang/ja/__init__.py in get_dtokens_and_spaces(dtokens, text, gap_tag)
    105         try:
--> 106             word_start = text[text_pos:].index(word)
    107         except ValueError:

ValueError: substring not found

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-7-1ecb66ffae52> in <module>
      4 sp = spacy.load('ja_core_news_sm')
      5 t  = 'あつつ( ̄^ ̄)ゞ'
----> 6 sp(t)

/usr/local/virtualenv/tf1/lib/python3.6/site-packages/spacy/language.py in __call__(self, text, disable, component_cfg)
    439                 Errors.E088.format(length=len(text), max_length=self.max_length)
    440             )
--> 441         doc = self.make_doc(text)
    442         if component_cfg is None:
    443             component_cfg = {}

/usr/local/virtualenv/tf1/lib/python3.6/site-packages/spacy/lang/ja/__init__.py in make_doc(self, text)
    279 
    280     def make_doc(self, text):
--> 281         return self.tokenizer(text)
    282 
    283 

/usr/local/virtualenv/tf1/lib/python3.6/site-packages/spacy/lang/ja/__init__.py in __call__(self, text)
    143         sudachipy_tokens = self.tokenizer.tokenize(text)
    144         dtokens = self._get_dtokens(sudachipy_tokens)
--> 145         dtokens, spaces = get_dtokens_and_spaces(dtokens, text)
    146 
    147         # create Doc with tag bi-gram based part-of-speech identification rules

/usr/local/virtualenv/tf1/lib/python3.6/site-packages/spacy/lang/ja/__init__.py in get_dtokens_and_spaces(dtokens, text, gap_tag)
    106             word_start = text[text_pos:].index(word)
    107         except ValueError:
--> 108             raise ValueError(Errors.E194.format(text=text, words=words))
    109 
    110         # space token

ValueError: [E194] Unable to aligned mismatched text 'あつつ( ̄^ ̄)ゞ' and words '['あ', 'つつ', '(', ' ', '̄', '^', ' ̄', ')', 'ゞ']'.
```

**Comment by kovalevfm** [comment]
Apparently it's fixed in 2.3.4

**Comment by svlandeg** [comment]
Ah, if it's fixed in 2.3.4, I think we can go ahead and close this?

**Comment by kovalevfm** [comment]
Yes, you can close it, update have fixed the problem.
